packages/koa-performance/src/middleware.ts

```diff
@@ -0,0 +1,243 @@
+import {Context} from 'koa';
+import compose from 'koa-compose';
+import bodyParser from 'koa-bodyparser';
+import {snakeCase} from 'change-case';
+import {StatusCode, Header} from '@shopify/network';
+import {
+  EventType,
+  Navigation,
+  LifecycleEvent,
+  NavigationDefinition,
+  NavigationMetadata,
+} from '@shopify/performance';
+import {StatsDClient, Logger} from '@shopify/statsd';
+
+import {LifecycleMetric, NavigationMetric} from './enums';
+import {BrowserConnection} from './types';
+
+export interface Options {
+  prefix: string;
+  development: boolean;
+  statsdHost?: string;
+  statsdPort?: number;
+  anomalousNavigationDurationThreshold?: number;
+  logger?: Logger;
+  additionalTags?(
+    metricsBody: Metrics,
+    userAgent: string,
+  ): {[key: string]: string | number | boolean};
+  additionalNavigationTags?(
+    navigation: Navigation,
+  ): {[key: string]: string | number | boolean};
+  additionalNavigationMetrics?(
+    navigation: Navigation,
+  ): {name: string; value: any}[];
+}
+
+export interface Metrics {
+  pathname: string;
+  connection: Partial<BrowserConnection>;
+  events: LifecycleEvent[];
+  navigations: {
+    details: NavigationDefinition;
+    metadata: NavigationMetadata;
+  }[];
+}
+
+export enum TagValue {
+  ReallyFastConnection = 'Really fast',
+  Unknown = 'Unknown',
+}
+
+export function clientPerformanceMetrics({
+  statsdHost,
+  statsdPort,
+  prefix,
+  development,
+  logger,
+  anomalousNavigationDurationThreshold,
+  additionalTags: getAdditionalTags,
+  additionalNavigationTags: getAdditionalNavigationTags,
+  additionalNavigationMetrics: getAdditionalNavigationMetrics,
+}: Options) {
+  return compose([
+    bodyParser(),
+    async function clientPerformanceMetricsMiddleware(ctx: Context) {
+      const statsLogger: Logger = logger || ctx.state.logger;
+
+      const {body} = ctx.request as any;
```

Comment:
I guess now that we're bundling the bodyparser we could maybe type this better, but the function below narrows the type for body and the rest of our uses of `ctx` are strictly typed so it doesn't add much value regardless